plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XLanguageServerImpl.java

```diff
@@ -348,24 +371,6 @@ protected ServerCapabilities createServerCapabilities(InitializeParams params) {
 	@Override
 	public void initialized(InitializedParams params) {
 		clientInitialized.complete(params);
-		clientInitialized.join();
```

Comment:
The join call here was surprising me given that the future was completed in the line above on the same thread.
Since we want to start something immediately after the client was initialized, I figured that a `.thenRun(..)` on the future might be the reasonable thing to do.